File: Backend/app_administracion_medicamentos/src/main/java/app_administracion_medicamentos_backend/api/interfaces/services/IMedicamentoService.java
package app_administracion_medicamentos_backend.api.interfaces.services;

import java.util.List;



public interface IMedicamentoService<T> {

	public List<T> getAll();
	
	public T getById(Long id);
	
	public T createOrUpdate(T entity);
	
	public T deleteById(Long id);
	
	public List<T> getAllByClienteId(Long id_cliente);
	
}

File: Frontend/material-dashboard-angular2/src/app/programming/interfaces/IMedicamentoService.ts
import { Observable } from "rxjs";
import { Response } from "../models/Response.dto";

export interface IMedicamentoService<T>{

    getAll(): Observable<Response<T[]>>;

    getById(id: number): Observable<Response<T>>;

    createOrUpdate(entity: T): Observable<Response<T>>;

    deleteById(id: number): Observable<Response<T>>;

}